internal/impl/ollama/embeddings_processor_ollama.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/blob/main/licenses/rcl.md
+
+package ollama
+
+import (
+	"context"
+	"errors"
+	"unicode/utf8"
+
+	"github.com/ollama/ollama/api"
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	oepFieldText = "text"
+)
+
+func init() {
+	err := service.RegisterBatchProcessor(
+		"ollama_embeddings",
+		ollamaEmbeddingProcessorConfig(),
+		ollamaEmbeddingBatchProcessor,
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ollamaEmbeddingProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Fields(
+			service.NewStringField(bopFieldServerAddress).
+				Description("The address of a remote ollama server to use. If omitted a local ollama server will be started and ran.").
+				Optional(),
+			service.NewStringField(bopFieldModel).
+				Description("The ollama model to use").
+				Examples("nomic-embed-text", "mxbai-embed-large", "snowflake-artic-embed", "all-minilm"),
+			service.NewInterpolatedStringField(oepFieldText).
+				Description("The text to create embeddings for. By default the entire payload as a string is submitted.").
+				Optional(),
+		)
+}
+
+func ollamaEmbeddingBatchProcessor(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchProcessor, error) {
+	p := ollamaEmbeddingProcessor{}
+	if conf.Contains(oepFieldText) {
+		pf, err := conf.FieldInterpolatedString(oepFieldText)
+		if err != nil {
+			return nil, err
+		}
+		p.text = pf
+	}
+	b, err := newBaseProcessor(conf, mgr)
+	if err != nil {
+		return nil, err
+	}
+	p.baseOllamaProcessor = b
+	return &p, nil
+}
+
+type ollamaEmbeddingProcessor struct {
+	*baseOllamaProcessor
+
+	text *service.InterpolatedString
+}
+
+func (o *ollamaEmbeddingProcessor) ProcessBatch(ctx context.Context, batch service.MessageBatch) ([]service.MessageBatch, error) {
+	return o.baseOllamaProcessor.ProcessBatch(ctx, batch, func(ctx context.Context, msg *service.Message) (*service.Message, error) {
+		p, err := o.computeText(msg)
+		if err != nil {
+			return nil, err
+		}
+		e, err := o.generateEmbedding(ctx, p)
+		if err != nil {
+			return nil, err
+		}
+		m := service.NewMessage(nil)
+		m.SetStructuredMut(e)
```

Comment:
This is a bit annoying, but if you set `[]float64` in here, Bloblang won't be able to operate on it. You'll have to convert it to `[]any`. See docs here: https://pkg.go.dev/github.com/redpanda-data/benthos/v4@v4.32.0/public/service#Message.SetStructured

Reply:
Thank you! 🥇 I just ran into this today with the pinecone output and it was driving me bonkers why I couldn't get the length of this vector